### PR TITLE
added comments in circuit

### PR DESCRIPTION
**Pull request by ananas-block**

### CODE REVIEW

File: psp-examples/swap/README.md [hunk 1]
```diff
@@ -33,3 +33,63 @@ yarn test
...
+## Tutorial
+
+In this tutorial you will build a private swap which can be used to negotiate and settle an over the counter (OTC) transaction.
+
+We will implement a private solana program (PSP).
+The PSP repository 
+
+### Primer on Light Protocol:
+- Shielded Balance:
+    - Sield(deposit): You can deposit(shield) value to Light Protocol. You transfer to the Light liquidity pool and receive a utxo in return.
```
Comment (sergeytimoshin) [inline psp-examples/swap/README.md:47]: typo: "Sield"

File: psp-examples/swap/tests/swaps.ts [hunk 2]
```diff
@@ -90,24 +97,40 @@ describe("Test swaps", () => {
...
 
-  it("Swap Take functional", async () => {
+  /**
+   * 1. Create seller and buyer Users
+   * 2. seller user creates offer
+   *    - creates utxo
+   *    - encrypts it to the buyer
+   *    - stores the encrypted utxo onchain in a compressed account
+   * 3. recipient decrypts offer
+   * 4. recipient generates
```
Comment (sergeytimoshin) [inline psp-examples/swap/tests/swaps.ts:107]: "4. recipient generates" ...

File: psp-examples/swap/tests/swaps.ts [hunk 3]
```diff
@@ -271,6 +306,13 @@ describe("Test swaps", () => {
       completePspProofInputs,
       false,
     );
+    
+    /**
+     * Step 5:
+     * Create solana transactions.
+     */
```
Comment (vadorovsky) [inline psp-examples/swap/tests/swaps.ts:313]: You could merge this comment with the one below.

File: psp-examples/swap/tests/swaps.ts [hunk 4]
```diff
@@ -636,31 +586,98 @@ describe("Test swaps", () => {
...
     fetchedOfferUtxo.publicKey = STANDARD_SHIELDED_PUBLIC_KEY;
     offerUtxo.index = fetchedOfferUtxo.index;
-    Utxo.equal(POSEIDON, offerUtxo, fetchedOfferUtxo); // , false, sellerUser.account, buyerUser.account
+    Utxo.equal(POSEIDON, offerUtxo, fetchedOfferUtxo);
 
     console.log(
       `Successfully fetched and decrypted offer: priceSol ${fetchedOfferUtxo.appData.priceSol.toString()}, offer sol amount: ${fetchedOfferUtxo.amounts[0].toString()} \n recipient public key: ${fetchedOfferUtxo.appData.recipient.toString()}`,
     );
+    /**
+     * Offer trade 1 sol for 2 sol
```
Comment (vadorovsky) [inline psp-examples/swap/tests/swaps.ts:607]: Maybe let's swap SOL with SPL. Swapping SOL with SOL kinda doesn't make sense.

File: psp-examples/swap/tests/swaps.ts [hunk 5]
```diff
@@ -284,17 +344,24 @@ describe("Test swaps", () => {
...
     });
 
     /**
-     * Proves PSP logic
-     * returns proof and it's public inputs
+     * Generates the PSP proof.
+     * The PSP proof proves the PSP logic.
+     * In this case it enforces the constraints that an offer utxo
+     * can only be spent if a reward utxo exists for which the offer utxo data matches.
      */
```
Comment (sergeytimoshin) [inline psp-examples/swap/tests/swaps.ts:364]: /**
 * Generates the proof for the Private Solana Program (PSP).
 * This proof enforces the constraints ensuring an offer UTXO can only be spent 
 * if a corresponding reward UTXO exists with matching offer UTXO data.
 */

**Pull request closed without merging**